# origin: marcelo (BitKeeper)
# cset: 1.1583 (2.4) key=42388250emER3koZtkQ3YSiP7eNDpg
# URL: http://linux.bkbits.net:8080/linux-2.4/cset@42388250emER3koZtkQ3YSiP7eNDpg
# inclusion: upstream
# descrition: Paul Mackerras: Remote Linux DoS on ppp servers (CAN-2005-0384)
# revision date: Wed, 23 Mar 2005 13:47:45 +0900
#
# S rset: ChangeSet|1.1582..1.1583
# I rset: drivers/net/ppp_async.c|1.9..1.10
#
# Key:
# S: Skipped  ChangeSet file only
# O: Original Followed by Updated
# U: Updated  Included with updated range of versions
# I: Included Included verbatim
# E: Excluded Excluded on request from user
# D: Deleted  Manually deleted by subsequent user edit
# R: Revised  Manually revised by subsequent user edit
#
#
# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/03/16 13:35:16-03:00 marcelo@logos.cnet 
#   Paul Mackerras: Remote Linux DoS on ppp servers (CAN-2005-0384)
# 
# drivers/net/ppp_async.c
#   2005/03/16 13:34:30-03:00 marcelo@logos.cnet +1 -1
#   Paul Mackerras: Remote Linux DoS on ppp servers (CAN-2005-0384)
# 
#
===== drivers/net/ppp_async.c 1.9 vs 1.10 =====
--- 1.9/drivers/net/ppp_async.c	2004-12-16 22:57:23 +09:00
+++ 1.10/drivers/net/ppp_async.c	2005-03-17 01:34:30 +09:00
@@ -996,7 +996,7 @@ static void async_lcp_peek(struct asyncp
 	data += 4;
 	dlen -= 4;
 	/* data[0] is code, data[1] is length */
-	while (dlen >= 2 && dlen >= data[1]) {
+	while (dlen >= 2 && dlen >= data[1] && data[1] >= 2) {
 		switch (data[0]) {
 		case LCP_MRU:
 			val = (data[2] << 8) + data[3];
